.pre-commit-config.yaml

```diff
@@ -27,22 +27,22 @@ repos:
       - id: pydocstyle
 
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.1.0
+    rev: v4.2.0
     hooks:
       - id: check-toml
       - id: check-yaml
       - id: end-of-file-fixer
       - id: trailing-whitespace
 
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.812
+    rev: v0.942
     hooks:
       - id: mypy
         exclude: '^(docs|tasks|tests)|setup\.py'
         args: [--ignore-missing-imports]
```

Comment:
we have two options:
1. set only this to only version v0.812
or 

2.Lets remove this bit from pre-commit
if we would like to include, mypy
lets do it as separate check.
As i tried to resolve this and it seems there are many changes to be done on the source code to satisfy code.

<details>
        <summary>For mypy changes: Click here</summary>
```
src/models/predict_model.py:6:1: error: Function is missing a type annotation  [no-untyped-def]
test_environment.py:6:1: error: Function is missing a return type annotation  [no-untyped-def]
test_environment.py:6:1: note: Use "-> None" if function does not return a value
test_environment.py:26:5: error: Call to untyped function "main" in typed context  [no-untyped-call]
models/github-ttm/GitHubTTMModel.py:19:5: error: Function is missing a return type annotation  [no-untyped-def]
models/github-ttm/GitHubTTMModel.py:19:5: note: Use "-> None" if function does not return a value
models/github-ttm/GitHubTTMModel.py:27:22: error: Call to untyped function "load_model_from_s3" in typed context  [no-untyped-call]
models/github-ttm/GitHubTTMModel.py:29:5: error: Function is missing a return type annotation  [no-untyped-def]
models/github-ttm/GitHubTTMModel.py:56:58: error: Missing type parameters for generic type "Dict"  [type-arg]
models/github-ttm/GitHubTTMModel.py:56:65: error: Incompatible default for argument "meta" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
models/github-ttm/GitHubTTMModel.py:57:28: error: Missing type parameters for generic type "List"  [type-arg]
models/github-ttm/GitHubTTMModel.py:61:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:113:29: error: Class cannot subclass "BaseEstimator" (has type "Any")  [misc]
src/features/build_features.py:113:44: error: Class cannot subclass "TransformerMixin" (has type "Any")  [misc]
src/features/build_features.py:116:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:135:26: error: Call to untyped function "_finditem" in typed context  [no-untyped-call]
src/features/build_features.py:137:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:142:24: error: Call to untyped function "_finditem" in typed context  [no-untyped-call]
src/features/build_features.py:144:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:148:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:153:34: error: Class cannot subclass "BaseEstimator" (has type "Any")  [misc]
src/features/build_features.py:153:49: error: Class cannot subclass "TransformerMixin" (has type "Any")  [misc]
src/features/build_features.py:156:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:160:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:164:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:191:30: error: Class cannot subclass "BaseEstimator" (has type "Any")  [misc]
src/features/build_features.py:191:45: error: Class cannot subclass "TransformerMixin" (has type "Any")  [misc]
src/features/build_features.py:194:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:200:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:209:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:213:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:218:42: error: Call to untyped function "_directories_from_filepaths" of "ChangeInDirTransformer" in typed context  [no-untyped-call]
src/features/build_features.py:223:34: error: Class cannot subclass "BaseEstimator" (has type "Any")  [misc]
src/features/build_features.py:223:49: error: Class cannot subclass "TransformerMixin" (has type "Any")  [misc]
src/features/build_features.py:226:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:230:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:234:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:240:28: error: Class cannot subclass "BaseEstimator" (has type "Any")  [misc]
src/features/build_features.py:240:43: error: Class cannot subclass "TransformerMixin" (has type "Any")  [misc]
src/features/build_features.py:243:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:247:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:251:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:257:29: error: Class cannot subclass "BaseEstimator" (has type "Any")  [misc]
src/features/build_features.py:257:44: error: Class cannot subclass "TransformerMixin" (has type "Any")  [misc]
src/features/build_features.py:260:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:264:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:268:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:274:32: error: Class cannot subclass "BaseEstimator" (has type "Any")  [misc]
src/features/build_features.py:274:47: error: Class cannot subclass "TransformerMixin" (has type "Any")  [misc]
src/features/build_features.py:277:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:286:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:299:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:303:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:307:28: error: Call to untyped function "_get_filetype" in typed context  [no-untyped-call]
src/features/build_features.py:326:33: error: Class cannot subclass "BaseEstimator" (has type "Any")  [misc]
src/features/build_features.py:326:48: error: Class cannot subclass "TransformerMixin" (has type "Any")  [misc]
src/features/build_features.py:329:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:338:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:342:5: error: Function is missing a type annotation  [no-untyped-def]
src/features/build_features.py:361:1: error: Function is missing a type annotation  [no-untyped-def]
src/models/train_model.py:11:1: error: Function is missing a type annotation  [no-untyped-def]
src/models/train_model.py:23:26: error: Call to untyped function "standardize" in typed context  [no-untyped-call]
src/models/train_model.py:85:1: error: Function is missing a type annotation  [no-untyped-def]
src/data/make_dataset.py:17:1: error: Function is missing a type annotation  [no-untyped-def]
src/data/make_dataset.py:27:1: error: Function is missing a type annotation  [no-untyped-def]
src/data/make_dataset.py:33:18: error: Call to untyped function "testgrid_labelwise_encoding" in typed context  [no-untyped-call]
src/data/make_dataset.py:54:1: error: Function is missing a type annotation  [no-untyped-def]
src/data/make_dataset.py:69:1: error: Function is missing a type annotation  [no-untyped-def]
src/data/make_dataset.py:107:34: error: Call to untyped function "decode_run_length" in typed context  [no-untyped-call]
src/data/make_dataset.py:202:1: error: Function is missing a type annotation  [no-untyped-def]
models/optimal-stopping-point/OSPModel.py:28:5: error: Function is missing a return type annotation  [no-untyped-def]
models/optimal-stopping-point/OSPModel.py:28:5: note: Use "-> None" if function does not return a value
models/optimal-stopping-point/OSPModel.py:38:28: error: Call to untyped function "init_s3_resource" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:40:5: error: Function is missing a return type annotation  [no-untyped-def]
models/optimal-stopping-point/OSPModel.py:58:58: error: Missing type parameters for generic type "Dict"  [type-arg]
models/optimal-stopping-point/OSPModel.py:58:65: error: Incompatible default for argument "meta" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
models/optimal-stopping-point/OSPModel.py:59:28: error: Missing type parameters for generic type "List"  [type-arg]
models/optimal-stopping-point/OSPModel.py:65:18: error: Missing type parameters for generic type "Dict"  [type-arg]
models/optimal-stopping-point/OSPModel.py:88:23: error: Call to untyped function "fetch_all_tests" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:89:22: error: Call to untyped function "fetch_all_tests" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:92:25: error: Call to untyped function "filter_test_type" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:96:24: error: Call to untyped function "filter_test_type" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:102:36: error: Call to untyped function "fit_distribution" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:106:41: error: Call to untyped function "best_distribution" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:109:35: error: Call to untyped function "fit_distribution" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:119:51: error: Call to untyped function "standardize" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:122:51: error: Call to untyped function "standardize" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:127:15: error: Call to untyped function "optimal_stopping_point" in typed context  [no-untyped-call]
models/optimal-stopping-point/OSPModel.py:137:9: error: Returning Any from function declared to return "float"  [no-any-return]

```
</details>

Reply:
reverted this back to v0.812. Should this work @harshad16 ?